### DISCUSSION TITLE
Default rmode and bgcolor in Startup.cs (ImageSharp.Web)

### DISCUSSION THREAD

**Discussion by sjorritsma** [SixLabors/ImageSharp.Web#177] [Q&A]
Hi thanks for your work, LOVE IT!

Can't figure this out. I would like to set default Rmode and Bgcolor in the Startup.cs.

Managed to set default quality to 80

```c#
var configuration = new Configuration(
                new PngConfigurationModule(),
                new GifConfigurationModule(),
                new BmpConfigurationModule(),
                new TgaConfigurationModule());

          
            var jpegEncoder = new JpegEncoder { Quality=80};
            configuration.ImageFormatsManager.SetEncoder(JpegFormat.Instance, jpegEncoder);
            configuration.ImageFormatsManager.SetDecoder(JpegFormat.Instance, new JpegDecoder());
            configuration.ImageFormatsManager.AddImageFormatDetector(new JpegImageFormatDetector());
            return configuration;


services.AddImageSharp(
                 options =>
                 {
                     options.Configuration = CreateCustomConfiguration();
                     options.BrowserMaxAge = TimeSpan.FromDays(90);
                     options.CacheMaxAge = TimeSpan.FromDays(365);
                 }
                );
```

**Comment by JimBobSquarePants** [comment]
Glad to hear you like it!

That's not something you can do via the Configuration class. Quality is a property of a codec not a process (we actually replace the default encoder on a per request basis when parsing commands)

You could set automatic commands via [ImageSharpMiddlewareOptions.OnParseCommandsAsync](https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddlewareOptions.html#SixLabors_ImageSharp_Web_Middleware_ImageSharpMiddlewareOptions_OnParseCommandsAsync) though which could be used to set a background or resize mode.
